server/mediaserver.go

```diff
@@ -509,10 +537,121 @@ func getRTMPStreamHandler(s *LivepeerServer) func(url *url.URL) (stream.RTMPVide
 
 //End RTMP Handlers
 
+func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
+	// we read this unconditionally, mostly for ffmpeg
+	body, err := ioutil.ReadAll(r.Body)
+
+	if err != nil {
+		http.Error(w, fmt.Sprintf(`Error reading http request body: %s`, err.Error()), http.StatusInternalServerError)
+		return
+	}
+	r.Body.Close()
+	r.URL = &url.URL{Scheme: "http", Host: r.Host, Path: r.URL.Path}
+
+	if ".ts" != path.Ext(r.URL.Path) {
+		// ffmpeg sends us a m3u8 as well, so ignore
+		// Alternatively, reject m3u8s explicitly and take any other type
+		// TODO also look at use content-type
+		http.Error(w, fmt.Sprintf(`ignoring file extension: %s`, path.Ext(r.URL.Path)), http.StatusBadRequest)
+		return
+	}
+
+	now := time.Now()
+	mid := parseManifestID(r.URL.Path)
+	s.connectionLock.Lock()
+	cxn, exists := s.rtmpConnections[mid]
+	if exists && cxn != nil {
+		cxn.lastUsed = now
+	}
+	s.connectionLock.Unlock()
+
+	// Check for presence and register if a fresh cxn
+	if !exists {
+		appData := (createRTMPStreamIDHandler(s))(r.URL)
+		if appData == nil {
+			http.Error(w, "Could not create stream ID: ", http.StatusInternalServerError)
+			return
+		}
+		st := stream.NewBasicRTMPVideoStream(appData)
+		cxn, err = s.registerConnection(st)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		ticker := time.NewTicker(RefreshIntervalHttpPush)
+
+		go func(s *LivepeerServer, mid core.ManifestID) {
+			for _ = range ticker.C {
+				s.connectionLock.RLock()
+				lastUsed := s.rtmpConnections[mid].lastUsed
+				s.connectionLock.RUnlock()
+
+				if time.Since(lastUsed) > RefreshIntervalHttpPush {
+					_ = removeRTMPStream(s, mid)
+					ticker.Stop()
+					return
+				}
+			}
+		}(s, mid)
+	}
+
+	fname := path.Base(r.URL.Path)
+	seq, err := strconv.ParseUint(strings.TrimSuffix(fname, path.Ext(fname)), 10, 64)
+	if err != nil {
+		seq = 0
```

Comment:
So, if names in incoming segments would be not parseable, every segment would be having `SeqNo` set to zero, and playlist will always contains just one segment. Maybe shouldn't accept segments for which we couldn't find `SeqNo`?

Reply:
That's a good point @darkdarkdragon. @j0sh do you have any color on this one?

Reply:
Probably OK. I can see cases where we might want to be more lenient wrt naming but we don't have to cover those in this PR.

Reply:
Probably OK as it is now, or OK to to change it to not accept segments with un-parsable names?

Reply:
Either works for me